src/rgw/rgw_asio_frontend.cc

```diff
@@ -134,6 +136,10 @@ void handle_connection(boost::asio::io_context& context,
                        boost::system::error_code& ec,
                        spawn::yield_context yield)
 {
+
+  std::call_once(tracerInit,[](){
+    tracer_2.init_tracer("RGW_Client_Process", "../src/tracerConfig.yaml");
```

Comment:
is this something happening in runtime?
where is this yaml supposed to be?

Reply:
yes `tracer_2` needs to be initialized only once during run time.
The yaml file relative path is given so I think it will be fine.

Reply:
thing the needs to be initialized once, should be initialized in the rgw main function.
this is to control the order of initialization, as well as the ability to change configuration of the singleton.

regarding the path, it is relative to what?
is this path used in compile time or during run time?

Reply:
yes happening in run time the path for config file is relative it is inside `src/` direc

Reply:
than this is wrong. the rgw binary could be installed in different places, which, for sure has nothing to do with the "src" directory.
the location of the tracer config should probably be a config parameter to RGW, together with the to turn tracing on and off.

Reply:
I tried initialiizing tracer in rgw_main but its not invoking the trace then, for now I will keep it in rgw_asio_frontend.cc

Reply:
the tracer is generic, unrelated to S3 requests, so, it should be in rgw_main. you have to debug and see why it does not work there.
sldo, even in the context of S3 requests, don't tie that to a specific frontend (there is also civerweb).

Reply:
The reason for this is once the main methods ends, the tracer destructor gets fired and it is closed, but in rgw_asio_frontend.cc my tracer is getting is initialized when a new request comes in and it remain initialized after every request has been completed i.e untill `end of stream` is not encountered. I was told civetweb is not that important and only `boost` frontend is required to so I put my initialization code inside this.

Reply:
its not about the importance of civerweb, you have a singleton that is initialized is a specific code flow.
if someone want to use it for some other flow in the rgw, it would create a problem for them.
anything that is global to the rgw should be in rgw_main

Reply:
okay I will see what I can do with this but Im not sure if I will be able to make it work

Reply:
please see: https://github.com/ceph/ceph/pull/35860/files/32cc6aaa6347048dfe9a72860da6dfc52506ab29#r449513443